src/jdk.jfr/share/classes/jdk/jfr/internal/settings/ThrottleSetting.java

```diff
@@ -94,7 +94,7 @@ public void setValue(String value) {
             // if unit is less than 1 s, scale samples
             if (unit.nanos < SECONDS.nanos) {
                 long perSecond = SECONDS.nanos / unit.nanos;
-                samples *= Utils.multiplyOverflow(samples, perSecond, Long.MAX_VALUE);
+                samples = Utils.multiplyOverflow(samples, perSecond, Long.MAX_VALUE);
```

Comment:
Reasoning from the math alone, it looks obvious that `samples *= samples` is not correct. So that part looks fine.

Thinking about `unit.nanos == SECONDS.nanos` case. This means the unit is exactly "seconds", so no adjustment is needed?

Reply:
Native expects the value in milliseconds, and the default value is 1000 ms, so it seems fine to me.